onnx_chainer/functions/array.py

```diff
@@ -52,10 +52,85 @@ def convert_Depth2Space(func, onnx_op_name, opset_version, input_names,
 
 def convert_GetItem(func, onnx_op_name, opset_version, input_names,
                     output_names, parameters):
-    return helper.make_node(
-        onnx_op_name, input_names, output_names,
-        slice=func.slices
-    ),
+    x = func.inputs[0]
+    axes, starts, ends = [], [], []
+    squeeze_idxs, unsqueeze_idxs = [], []
+    skipped = 0  # when set ellipsis, need to skip index rolling
+
+    for i, idx in enumerate(func.slices):
+        # axis means the index of input x, adjust None and Ellipsis counts
+        axis = i - len(unsqueeze_idxs) + skipped
+        if isinstance(idx, slice):
+            if idx.step is not None and idx.step != 1:
+                raise ValueError(
+                    'GetItem with {}step slicing is not supported in ONNX '
+                    'Slice operator'.format(idx.step))
+            axes.append(axis)
+            starts.append(idx.start)
+            ends.append(idx.stop)
+        elif isinstance(idx, int):
+            axes.append(axis)
+            starts.append(idx)
+            ends.append(idx+1)
+            squeeze_idxs.append(axis)
+        elif idx is None:
+            unsqueeze_idxs.append(i - len(squeeze_idxs) + skipped)
+        elif idx is Ellipsis:
+            # calculate rest slice number except None, GetItem does not allow
+            # multiple Ellipsis, so ignore latter Ellipsis count
+            rest_slice_len = len(
+                [idx_ for idx_ in func.slices[i+1:] if idx_ is not None])
+            for ellipsis_i in range(axis, len(x.shape)-rest_slice_len):
+                axes.append(ellipsis_i)
+                starts.append(0)
+                ends.append(x.shape[ellipsis_i])
+                skipped += 1
+            skipped -= 1  # for next loop
+        else:
+            # not support advanced index like `array[[0,1], [0, 1]]`
+            raise ValueError(
+                'GetItem with type {} cannot handle in ONNX Slice, so that '
```

Comment:
I guess this rejects `chainer.Variable(np.array([1,2,3]))[np.array(1)]`, which can be implemented by ONNX's `Slice`? I'm not sure if it's worth supporting though..

posted in proxy, https://github.com/chainer/onnx-chainer/pull/72#discussion_r254584914